server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -665,6 +662,25 @@ private void checkAliasNameConflicts(Map<String, String> renamedIndices, Set<Str
                         }
                     }
 
+                    private Set<String> applyAliasesWithRename(IndexMetadata snapshotIndexMetadata, IndexMetadata.Builder indexMdBuilder) {
+                        Set<String> aliases = new HashSet<>();
+                        Pattern renameAliasPattern = null;
+                        if (request.renameAliasPattern() != null && request.renameAliasReplacement() != null) {
+                            renameAliasPattern = Pattern.compile(request.renameAliasPattern());
+                        }
```

Comment:
I think you can improve on this slightly without the extra variables.

In the case this conditional fails, you're still iterating the map just to add the keys to another set to return.

How about reversing this conditional (checking if either is null) and if so just return the keyset directly.

Then if you're continuing on you know the rename pattern is not null and you can apply the rest of the logic.